jscomp/test/js_re_test.ml

```diff
@@ -1,12 +1,30 @@
 
 let suites = Mt.[
+  "matches", (fun _ ->
+    let re = [%re "/(\\d+)-(?:(\\d+))?/g"] in
+    let str = "3-" in
+    match re |> Js.Re.exec str with
+      | Some result ->
+        let defined = ref "" in
+        let _ = Js.Nullable.iter (Js.Re.matches result).(1) ((fun m ->
+          defined := m) [@bs]) in
+        let undefined = (Js.Re.matches result).(2) in
+        Eq(("3", Js.Nullable.null), (!defined, undefined))
+      | None -> Fail()
+  );
+
   "exec_literal", (fun _ ->
     match [%re "/[^.]+/"] |> Js.Re.exec "http://xxx.domain.com" with
-    | Some res -> 
-      Eq ("xxx", (res |> Js.Re.matches).(0) |> Js.String.substringToEnd ~from:7)
+    | Some res ->
+      (match Js.Nullable.to_opt (res |> Js.Re.matches).(0) with
```

Comment:
And similarly here, you shouldn't need to pattern match but can instead compare directly to `Js.Nullable.return "http://xxx"`